README.md

```diff
@@ -1,64 +1,199 @@
+1. If your account is present in more than one Azure AD tenant, select your profile at the top right corner in the menu on top of the page, and then **switch directory**.
+   Change your portal session to the desired Azure AD tenant.
+1. In the portal menu, select the **Azure Active Directory** service, and then select **App registrations**.
+
+#### Register the client app (active-directory-java-deviceprofile)
+
+1. Navigate to the Microsoft identity platform for developers [App registrations](https://go.microsoft.com/fwlink/?linkid=2083908) page.
+1. Click **New registration** on top.
+1. In the **Register an application page** that appears, enter your application's registration information:
+   - In the **Name** section, enter a meaningful application name that will be displayed to users of the app, for example `active-directory-java-deviceprofile`.
+   - Change **Supported account types** to **Accounts in any organizational directory**.
+1. Click on the **Register** button at the bottom to create the application.
+1. In the app's registration screen, find the **Application (client) ID** value and record it for later. You'll need it to configure the configuration file(s) later in your code.
+1. In the app's registration screen, click on the **Manifest** blade, and:
+   - In the manifest editor, set the ``allowPublicClient`` property to **true**
+   - Click on **Save** in the bar above the manifest editor.
+1. In the app's registration screen, click on the **Authentication** blade in the left and:
+   - In the **Suggested Redirect URIs for Public Clients(mobile, desktop)** select the check box of **https://login.microsoftonline.com/common/oauth2/nativeclient**.
+   - Click on **Save** in the bar above.
+1. In the app's registration screen, click on the **API permissions** blade in the left to open the page where we add access to the Apis that your application needs.
+   - Click the **Add a permission** button and then,
+   - Ensure that the **Microsoft APIs** tab is selected.
+   - In the *Commonly used Microsoft APIs* section, click on **Microsoft Graph**
+   - In the **Delegated permissions** section, select the **User.Read**, **User.ReadBasic.All** in the list. Use the search box if necessary.
+   - Click on the **Add permissions** button at the bottom.
+
+1. At this stage, the permissions are assigned correctly but since the client app does not allow users to interact, the user's themselves cannot consent to these permissions.
+   To get around this problem, we'd let the [tenant administrator consent on behalf of all users in the tenant](https://docs.microsoft.com/azure/active-directory/develop/v2-admin-consent).
+   Click the **Grant admin consent for {tenant}** button, and then select **Yes** when you are asked if you want to grant consent for the requested permissions for all account in the tenant.
+   You need to be the tenant admin to be able to carry out this operation.
```

Comment:
Is this step really necessary? If the tenant admin doesn't grant consent, won't the user get a chance to consent when performing the device code auth flow on the phone browser? (that's what we say in the scenario overview)

Reply:
Will Check and update

Reply:
This step is necessary

Reply:
Tested several times and now removed this step.

Reply:
changed the scope in the code. so this issue solved.